db/migrations/000010_fix_component_attributes_constraints.sql

```diff
@@ -0,0 +1,43 @@
+-- +goose NO TRANSACTION 
+-- +goose Up
+-- +goose StatementBegin
+ALTER TABLE versioned_attributes DROP CONSTRAINT IF EXISTS check_server_id_server_component_id;
+ALTER TABLE versioned_attributes ADD CONSTRAINT check_server_id_server_component_id CHECK (
+  (
+    (server_id is not null)::integer +
+    (server_component_id is not null)::integer
+  ) = 2
+);
+
+ALTER TABLE attributes DROP CONSTRAINT IF EXISTS check_server_id_server_component_id;
+ALTER TABLE attributes ADD CONSTRAINT check_server_id_server_component_id CHECK (
+  (
+    (server_id is not null)::integer +
+    (server_component_id is not null)::integer
+  ) = 2
```

Comment:
This should still be 1 as the attribute should be assigned to only a server or a server_component. Both should not be filled in.